Uchu.StandardScripts/Equipment/Coil.cs

```diff
@@ -0,0 +1,69 @@
+using System.Threading.Tasks;
+using Uchu.World;
+using Uchu.World.Scripting.Native;
+
+namespace Uchu.StandardScripts.Equipment
+{
+    public class Coil : ObjectScript
+    {
+        //there isn't a script for this in the client, but three scripts have basically the exact same code, so it's easier to do this
+        public int SkillID { get; set; }
+        public int CoilThreshold { get; set; }
+        private int CoilCount = 0;
+        protected bool Ready = false;
+        public Coil(GameObject gameObject) : base(gameObject)
+        {
+            Listen(gameObject.OnStart, () =>
+            {
+                if (gameObject is Item item)
+                {
+                    Listen(item.Owner.GetComponent<InventoryComponent>().OnUnEquipped, unequippedItem =>
+                    {
+                        if (unequippedItem == item)
+                        {
+                            CoilCount = 0;
+                            //remove recoil count when unequipped
+                        }
+                    });
+                    Listen(item.Owner.GetComponent<DestroyableComponent>().OnHealthChanged, (newH, delta) =>
+                    {
+                        if (delta < 0)
+                        {
+                            Process(item);
+                        }
+                    });
+                    Listen(item.Owner.GetComponent<DestroyableComponent>().OnArmorChanged, (newA, delta) =>
+                    {
+                        if (delta < 0)
+                        {
+                            Process(item);
+                        }
+                    });
+                }
+            });
+        }
+        protected void Process(Item item)
+        {
+            if (item.IsEquipped && Ready)
+            {
+                CoilCount++;
+                if (CoilCount >= CoilThreshold)
+                {
+                    CoilCount = 0;
+                    Effect(item.Owner);
+                }
+            }
+            else
+            {
+                CoilCount = 0;
+            }
+        }
+        private async void Effect(GameObject target)
+        {
+            //TODO: remove delay
+            Task.Delay(100);
```

Comment:
Regarding the TODO comment, what prevents this delay from being removed?

Reply:
It's the same as some of the other 100ms delays scattered around, I didn't get to changing the order of the methods for player stats yet